pkg/controller/storagecluster/kms_resources.go

```diff
@@ -0,0 +1,56 @@
+package storagecluster
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	ocsv1 "github.com/openshift/ocs-operator/pkg/apis/ocs/v1"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (r *ReconcileStorageCluster) ensureKMSSystem(instance *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	// if there is no 'Security' spec to inspect, return silently
+	if instance.Spec.Security == nil {
+		return nil
+	}
+	if instance.Spec.Security.KeyManagementService != nil &&
+		instance.Spec.Security.KeyManagementService.ConnectionDetails == "" {
```

Comment:
I'd extract the last condition and make it its own `if` statement, it's clearer/easier to read.

Reply:
Simplified this part. Now KMS will be enabled under `Encryption` spec (no need to pass the token or configmap names).
Spec yaml may look something like this. 

```yaml
encryption:
    enable: true
    kms:
        enable: true
```
When `kms` is enabled, operator will check for the predefined configmap and token names, and update the CephCluster accordingly.

@nbalacha ^^^